src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -147,6 +147,10 @@ public void configure(ActionMenuWidget actionMenu,
 		actionMenu.addControllerWidget(uploader.asWidget());
 		if (!isUserAuthenticated) {
 			actionMenu.setToolsButtonVisible(false);
+			if (permissions.getCanPublicRead()) {

```

Comment:
add a test to verify annotations and file history are configured when not logged in, and can public read (and not configured otherwise).